### DISCUSSION TITLE
Is there an example of a more complex UI test using Avalonia.Headless, where a running application is tested?

### DISCUSSION THREAD

**Discussion by bmscodespace** [AvaloniaUI/Avalonia#16086] [Q&A]
Hi to all ;)

I am looking for an example where say XUnit together with Avalonia.Headless is used to test a running application. So f.e. a login scenario is tested. I guess this sounds like an integration test ;). 

Thank you for any answer ...

**Comment by maxkatz6** [comment]
I have used headless testing for "login scenario" before. Do you have any specific questions?
From the e2e point of view, it's not really more complex than doing input into two text boxes and pressing the button.

Also see:
https://docs.avaloniaui.net/docs/concepts/headless/
https://github.com/AvaloniaUI/Avalonia.Samples/tree/main/src/Avalonia.Samples/Testing/TestableApp.Headless.XUnit

**Reply by bmscodespace** [reply]
Hi, thank you very much for your answer and especially for your example. 
I have a follow up to my question above:
What if inside our MainWindow we have a ContentControl where the application binds to different UserControls depending on the applications state?  So f.e. we have something like 

```
<window>
...
  <ContentControl Content="{Binding Content}" />
...
</window>
```

where 

```
public ReactiveObject Content
{
            get => _content;
            private set => this.RaiseAndSetIfChanged(ref _content, value);
}
```
and where f.e.

we set 

`Content = InterestingViewModel()`

How would then a test look like? I guess, one would initialize the test with something like 

```
var window = new MainWindow
{
       Content = new InterestingViewModel(....)
       DataContext = new MainWindowViewModel(.....)
};

window.show();
```
Correct?
